src/compiler/factoryPublic.ts

```diff
@@ -2463,64 +2463,152 @@ namespace ts {
+        return tag;
+    }
+
+    export function createJSDocEnumTag(typeExpression?: JSDocTypeExpression, comment?: string) {
+        const tag = createJSDocTag<JSDocEnumTag>(SyntaxKind.JSDocEnumTag, "enum", comment);
+        tag.typeExpression = typeExpression;
+        return tag;
+    }
+
+    export function createJSDocTemplateTag(constraint: JSDocTypeExpression | undefined, typeParameters: readonly TypeParameterDeclaration[], comment?: string) {
+        const tag = createJSDocTag<JSDocTemplateTag>(SyntaxKind.JSDocTemplateTag, "template", comment);
+        tag.constraint = constraint;
+        tag.typeParameters = asNodeArray(typeParameters);
+        return tag;
+    }
+
+    export function createJSDocTypedefTag(fullName?: JSDocNamespaceDeclaration | Identifier, name?: Identifier, comment?: string, typeExpression?: JSDocTypeExpression | JSDocTypeLiteral) {
+        const tag = createJSDocTag<JSDocTypedefTag>(SyntaxKind.JSDocTypedefTag, "typedef", comment);
+        tag.fullName = fullName;
+        tag.name = name;
+        tag.typeExpression = typeExpression;
+        return tag;
+    }
+
+    export function createJSDocCallbackTag(fullName: JSDocNamespaceDeclaration | Identifier | undefined, name: Identifier | undefined, comment: string | undefined, typeExpression: JSDocSignature) {
+        const tag = createJSDocTag<JSDocCallbackTag>(SyntaxKind.JSDocCallbackTag, "callback", comment);
+        tag.fullName = fullName;
+        tag.name = name;
+        tag.typeExpression = typeExpression;
+        return tag;
+    }
+
+    export function createJSDocSignature(typeParameters: readonly JSDocTemplateTag[] | undefined, parameters: readonly JSDocParameterTag[], type?: JSDocReturnTag) {
+        const tag = createSynthesizedNode(SyntaxKind.JSDocSignature) as JSDocSignature;
+        tag.typeParameters = typeParameters;
+        tag.parameters = parameters;
+        tag.type = type;
+        return tag;
+    }
+
+    function createJSDocPropertyLikeTag<T extends JSDocPropertyLikeTag>(kind: T["kind"], tagName: "arg" | "argument" | "param", typeExpression: JSDocTypeExpression | undefined, name: EntityName, isNameFirst: boolean, isBracketed: boolean, comment?: string) {
+        const tag = createJSDocTag<T>(kind, tagName, comment);
+        tag.typeExpression = typeExpression;
+        tag.name = name;
+        tag.isNameFirst = isNameFirst;
+        tag.isBracketed = isBracketed;
+        return tag;
+    }
+
+    export function createJSDocPropertyTag(typeExpression: JSDocTypeExpression | undefined, name: EntityName, isNameFirst: boolean, isBracketed: boolean, comment?: string) {
+        return createJSDocPropertyLikeTag<JSDocPropertyTag>(SyntaxKind.JSDocPropertyTag, "param", typeExpression, name, isNameFirst, isBracketed, comment);
+    }
+
+    export function createJSDocParameterTag(typeExpression: JSDocTypeExpression | undefined, name: EntityName, isNameFirst: boolean, isBracketed: boolean, comment?: string) {

```

Comment:
why have both `createJSDocParameterTag` and `createJSDocParamTag`. The only real difference is that this sets `isNameFirst` but `createJSDocParamTag` doesn't.

Naively, it feels to me like we should only have one, which uses this code but is named `createJSDocParamTag`.

Reply:
Well, except that we shouldn't break the API unnecessarily. Ugh. Can you add documentation saying that `createJSDocParamTag` is deprecated and that this is the preferred function now?